sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -81,7 +121,16 @@ case class Mode(
       return null
     }
 
-    buffer.maxBy(_._2)._1
+    (if (right.eval().asInstanceOf[Boolean]) {
```

Comment:
nit: we can follow `PercentileBase#percentages` and evaluate the foldable parameters in a lazy val.

Reply:
Sure. Fixed in latest.